MapboxNavigation/CarPlayNavigationViewController.swift

```diff
@@ -167,15 +169,18 @@ public class CarPlayNavigationViewController: UIViewController {
     public override func viewSafeAreaInsetsDidChange() {
         super.viewSafeAreaInsetsDidChange()
         
-        if let previousSafeAreaInsets = previousSafeAreaInsets {
-            let navigationBarIsOpen = view.safeAreaInsets > previousSafeAreaInsets
-            mapView?.compassView.isHidden = navigationBarIsOpen
+        defer {
+            previousSafeAreaInsets = view.safeAreaInsets
+            view.setNeedsUpdateConstraints()
         }
         
-        previousSafeAreaInsets = view.safeAreaInsets
+        // TODO: Verify `?? 0` should result in navigationBarIsOpen = false initially
+        let navigationBarIsOpen = view.safeAreaInsets.top > previousSafeAreaInsets?.top ?? 0
+        mapView?.compassView.isHidden = navigationBarIsOpen
         
         // Adjust the map’s vanishing point to counterbalance the side maneuver panels by extending the view off beyond the other side of the screen.
         if let mapView = mapView {
+            // TODO: We observe safeAreaInsets but safeArea is not updated initially
```

Comment:
Isn’t `safeArea` just a convenience for `safeAreaInsets` that accounts for iOS 11 and below? If this to-do is because we’d like to directly constrain to the map view’s `safeArea`, then we could add a dependency between the two properties via `keyPathsForValuesAffecting(_:)`.